eis_toolkit/spatial_analyses/cell_based_association.py

```diff
@@ -0,0 +1,573 @@
+import os
+from numbers import Number
+
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+import rasterio
+from beartype import beartype
+from beartype.typing import List, Optional, Tuple, Union
+from shapely import wkt
+from shapely.geometry import Polygon
+
+os.environ["USE_PYGEOS"] = "0"
+
+
+@beartype
+def cell_based_association(
+    cell_size: int,
+    geodata: List[gpd.GeoDataFrame],
+    output_path: str,
+    column: Optional[List[str]] = [""],
+    subset_target_attribute_values: Optional[List[Union[None, list]]] = [None],
+    add_name: Optional[List[Union[str, None]]] = [None],
+    add_buffer: Optional[List[Union[Number, bool]]] = [False],
+) -> gpd.GeoDataFrame:
+    """Creation of CBA matrix.
+
+    Initializes a CBA matrix from a vector file. The mesh is calculated
+    according to the geometries contained in this file and the size of cells.
+    Allows to add multiple vector data to the matrix, based on targeted shapes
+    and/or attributes.
+
+    Args:
+        cell_size: Size of the cells.
+        geodata: GeoDataFrame to create the CBA matrix. Additional
+            GeoDataFrame(s) can be imputed to add to the CBA matrix.
+        output_path: Name of the saved .tif file.
+        column: Name of the column of interest. If no attribute is specified,
+            then an artificial attribute is created representing the presence
+            or absence of the geometries of this file for each cell of the CBA
+            grid. If a target attribute is ndicated, it must be categorical.
```

Comment:
Typo here, "ndicated". Also, I think I get what the categorical attribute means here, but why could not a numerical or string type of attribute be categorical as is stated below? If the column has only for example values "1", "2", "3", or 1, 2, 3.

Reply:
By default, the dummification process does not work with the integer values. I added a step to the `_add_layer()` and `_init_from_vector_data()` functions to convert targeted column to the categorical dtype, so numerical values should be working too now.